firmware/ventilator-controller-stm32/Core/Inc/Pufferfish/Util/COBS.tpp

```diff
@@ -39,25 +39,37 @@ IndexStatus encode_cobs(
   size_t code_index = 0;
   uint8_t code = 1;
 
-  size_t encoded_buffer_size = 0;
-
-  if (buffer.size() == max_block_size) {
-    for (size_t i = 0; i < max_block_size; i++) {
-      if (buffer[i] == 0) {
-        encoded_buffer_size = max_block_size + 1;
-        break;
-      }
-      encoded_buffer_size = max_block_size + 2;
+  // calculate the exact encoded buffer size
```

Comment:
As discussed, let's decompose the size-computation logic out into a separate function.